core/src/main/java/org/jdbi/v3/Query.java

```diff
@@ -87,8 +89,18 @@
     }
 
     @Override
-    public <ContainerType> ContainerType collectInto(Class<ContainerType> containerType) {
-        return collect(getCollectorFactoryRegistry().createCollectorFor(containerType));
+    public <R> R collectInto(GenericType<R> containerType) {
+        return this.<R>collectInto(containerType.getType());
+    }
```

Comment:
I still think we should remove this method.

The `ResultType` parameter is erased at runtime, so we can't verify that `R` here is a valid signature for containing `ResultType`.

    handle.createQuery("select * from foo")
        .mapTo(Foo.class)
        .collectInto(new GenericType<ImmutableList<Bar>>() {});

This is a recipe for inexplicable `ClassCastException`s happening in client code after JDBI has returned.

@stevenschlansker I understand your desire to free the caller from having to know where the collector factories are, but I think the risk to type safety outweighs the convenience of this method.

Moreover, the code is actually shorter if you just use the `collect` method:

    handle.createQuery("select * from foo")
        .mapTo(Foo.class)
        .collect(GuavaCollectors.toImmutableList()); // <-- could be be statically imported too